src/api/spec/controllers/webui/request_controller/show_spec.rb

```diff
@@ -0,0 +1,360 @@
+require 'rails_helper'
+
+RSpec.describe Webui::RequestController, '#show', vcr: true do
+  let(:submitter) { create(:confirmed_user, :with_home, login: 'kugelblitz') }
+  let(:receiver) { create(:confirmed_user, :with_home, login: 'titan') }
+  let(:target_project) { receiver.home_project }
+  let(:target_package) { create(:package_with_file, name: 'goal', project_id: target_project.id) }
+  let(:source_project) { submitter.home_project }
+  let(:source_package) { create(:package, :as_submission_source, name: 'ball', project: source_project) }
+  let(:bs_request) do
+    create(:bs_request_with_submit_action,
+           description: 'Please take this',
+           creator: submitter,
+           target_package: target_package,
+           source_package: source_package)
+  end
+
+  describe 'GET #show' do
+    context 'when the user is part of the beta program' do
+      render_views
+
+      before do
+        Flipper.enable(:request_show_redesign, receiver)
+      end
+
+      context 'when the user has only one submit action' do
+        let!(:comment) { create(:comment, commentable: bs_request) }
+
+        before do
+          login receiver
+          get :show, params: { number: bs_request.number }
+        end
+
+        it 'shows the new redesign page' do
+          expect(response.body).to have_text("Request #{bs_request.number}")
+          expect(response.body).to have_text('beta program')
+        end
+      end
+
+      context 'when the request has an add maintainer request action'
+      context 'when the request has an add bugowner request action'
+      context 'when the request has a submit request action'
+      context 'when the request has a delete request action'
+      context 'when the request has a maintenance incident request action'
+      context 'when the request has a maintenance release request action'
+      context 'when the request has a set bugowner request action'
+
+      context 'when having an accepted review' do
```

Comment:
I wonder what the significance of this context is? Why do you test this?

Reply:
I test this, because when there are accepted reviews, we render the accordion at the bottom. I want to test that the review accordion is working.

Reply:
The [AccordionReviewsComponent](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/components/accordion_reviews_component.rb) could be tested directly without having to test it here in controller specs. This will involve less setup in the specs and should be a tiny bit faster.

Reply:
Yes, and no. Exhaustive testing should go into the AccordionReviewsComponent, yes. But we need to test the integration of the component with the partials also (if we need  a param in the component and somehow the caller don't reflect that in the partial the component specs won't catch that kind of error). Testing the "happy path" is enough for this.